plaso/engine/engine.py

```diff
@@ -317,17 +319,25 @@ def BuildFilterFindSpecs(
           'building find specification based on artifacts: {0:s}'.format(
               ', '.join(artifact_filter_names)))
 
-      artifacts_registry_object = cls.BuildArtifactsRegistry(
+      artifacts_registry_object = BaseEngine.BuildArtifactsRegistry(
           artifact_definitions_path, custom_artifacts_path)
-      artifact_filters_object = (
+      self._artifacts_filter_helper = (
           artifact_filters.ArtifactDefinitionsFilterHelper(
-              artifacts_registry_object, artifact_filter_names,
-              knowledge_base_object))
-      artifact_filters_object.BuildFindSpecs(
-          environment_variables=environment_variables)
-      find_specs = knowledge_base_object.GetValue(
-          artifact_filters_object.KNOWLEDGE_BASE_VALUE)[
-              artifact_types.TYPE_INDICATOR_FILE]
+              artifacts_registry_object, knowledge_base_object))
+      self._artifacts_filter_helper.BuildFindSpecs(
+          artifact_filter_names, environment_variables=environment_variables)
+
+      if self._artifacts_filter_helper.registry_find_specs:
```

Comment:
Please add a comment here about why this is being done. The idea here is if the user wants a registry artifact, they probably also want to include the registry hive files, right?

Also, doesn't this logic belong in the artifact filter helper, given this is about constructing artifact filters? Please move it there.

Reply:
Let's keep it here for now, and have a longer conversation about this. I'll add the comment.

Reply:
OK, https://github.com/log2timeline/plaso/issues/2438